onnxscript/version_converter/_adapter_lib.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Stores all operator schema related changes between opsets"""
+
+from __future__ import annotations
+
+from onnxscript import ir
+
+### Adapters
+
+
+# Adapter to convert from an attribute to a node input
+def adapter_attr_to_input(attr_name: str):
+    def adapt(node: ir.Node):
+        assert attr_name in node._attributes
+        _attr = node._attributes.pop(attr_name)
+        # Create an ir.Value obj with properties of the axis attribute
+        attr_as_input = ir.Value(
+            name=_attr.name,
+        )
+        # Add the ir.Value as inputs to the node and graph
+        node._inputs = node._inputs + (attr_as_input,)
```

Comment:
note: avoid using and modifying private fields. To change inputs to a node, always initialize a new node to replace the current one.

Reply:
Further, we can't just create a value as above ... we need to create a constant value with the given value. For now, the simplest way is to create a new `Constant` node. (Orthogonally to this, we should extend the "builder" API we currently use to create initializers as well, but that's a separate issue, for now Constant nodes should be fine.)